flink-runtime-web/web-dashboard/src/app/pages/job/job.component.ts

```diff
@@ -49,6 +52,9 @@ export class JobComponent implements OnInit, OnDestroy {
         takeUntil(this.destroy$),
         mergeMap(() =>
           this.jobService.loadJob(this.activatedRoute.snapshot.params.jid).pipe(
+            tap(job => {
+              this.jobLocalService.setJobDetail(job);
+            }),
```

Comment:
We don't need to resolve this here, but I think this is going the wrong way: the service should be providing the data and flow it into the components, not the component loading the data and pushing it into a service.

Reply:
This line of code was actually written in the API method's pipe. I moved it here in order to decouple it from the api service to a service which is only responsible for job's state.